vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -103,39 +117,42 @@
     register: shutdown_vms
     retries: 50
     delay: 10
-    until: WORKER_NODE_VM in shutdown_vms.list_vms
+    until: MASTER_NODE_1_VM and MASTER_NODE_2_VM in shutdown_vms.list_vms
     become: yes
     become_user: root
 
-  - name: Power on "{{ WORKER_NODE }}"
+  - name: Power on masters
     shell: |
-       kubectl annotate bmh "{{ WORKER_NODE }}" -n "{{ NAMESPACE }}" reboot.metal3.io/poweroff-
+       kubectl annotate bmh "{{ item }}" -n "{{ NAMESPACE }}" reboot.metal3.io/poweroff-
+    with_items:
+      - "{{ MASTER_NODE_1 }}"
+      - "{{ MASTER_NODE_2 }}"
 
-  - name: List only running VMs
-    virt:
-      command: list_vms
-      state: running
-    register: running_vms
-    retries: 50
-    delay: 10
-    until: WORKER_NODE_VM in running_vms.list_vms
-    become: yes
-    become_user: root
-
-  - name: Powered on node status
-    delegate_to: "{{ CLUSTER_APIENDPOINT_IP }}"
+  - name: Check the status of powered off master nodes
+    delegate_to: "{{ MASTER_0_IP.stdout }}"
     vars:
       ansible_user: "{{ IMAGE_USERNAME }}"
       ansible_ssh_private_key_file: "{{ SSH_PRIVATE_KEY }}"
       ansible_python_interpreter: /usr/bin/python3
     become: yes
     become_user: "{{ IMAGE_USERNAME }}"
     block:
```

Comment:
why a block here if you have only one item in it ?

Reply:
Fixed.